### PR TITLE
Cover CompatibilityValidator with unit tests

### LINKED ISSUE #1148: Refactor GraphDatabaseConfiguration and improve test coverage
It's currently very difficult to test [GraphDatabaseConfiguration Class](https://github.com/chupman/janusgraph/blob/master/janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java#L1235)

Once it's refactored the upgrade feature from PR #998 needs test coverage.

### PR DESCRIPTION
**Pull request by porunov**
Signed-off-by: Oleksandr Porunov <alexandr.porunov@gmail.com>

This PR is related to a flow to improve test coverage of the old `GraphDatabaseConfiguration`.
Related to #1148 and #1347

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by chupman** [comment]
Hey @porunov I'm going to hold off on reviewing this for now since I anticipate the upgrade to junit 5, #1327, going in mid next week, and these tests will have to be rewritten slightly. Since it's a moderately significant change I'll post something in janusgraph-dev to let everyone know to expect it.

**Comment by porunov** [comment]
@chupman Sure, I agree with you that #1327 should be merged first. I will also try to review that PR these weekends.

File: janusgraph-test/src/main/java/org/janusgraph/graphdb/configuration/validator/CompatibilityValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 JanusGraph Authors
```
Comment (farodin91) [inline janusgraph-test/src/main/java/org/janusgraph/graphdb/configuration/validator/CompatibilityValidatorTest.java]: This file should in the folder`/janusgraph-test/src/test/java/org/janusgraph/graphdb/configuration/validator`. 
As you can see in the travis ci details, see for example here https://travis-ci.org/JanusGraph/janusgraph/jobs/487161233.

Comment (farodin91) [reply janusgraph-test/src/main/java/org/janusgraph/graphdb/configuration/validator/CompatibilityValidatorTest.java]: In `main` are tests which should run against different backends or utilities and in `test` are actual tests which are independently from any backend. These tests are executed in Travis ci.

Comment (porunov) [reply janusgraph-test/src/main/java/org/janusgraph/graphdb/configuration/validator/CompatibilityValidatorTest.java]: Thanks for clarifying! Now I get it.

**Comment by porunov** [comment]
@chupman I've rebased this PR against the latest master commit. Will you be able to make a review?

@farodin91 I've relocated the file where you said.

**Comment by porunov** [comment]
@chupman I think this PR can be merged by following lazy consensus.

**Comment by porunov** [comment]
@chupman ping

**Pull request merged**